src/Http/Controllers/LogViewerController.php

```diff
@@ -141,14 +141,24 @@ public function showByLevel(Request $request, $date, $level)
     public function search(Request $request, $date, $level = 'all')
     {
         $query   = $request->get('query');
-
+        
         if (is_null($query))
             return redirect()->route($this->showRoute, [$date]);
 
+        $needles = explode(' ', $query);
+
         $log     = $this->getLogOrFail($date);
         $levels  = $this->logViewer->levelsNames();
-        $entries = $log->entries($level)->filter(function (LogEntry $value) use ($query) {
-            return Str::contains($value->header, $query);
+        $entries = $log->entries($level)->filter(function ( LogEntry $value) use ($needles) {
```

Comment:
Revert the changes to this;
```php
$query   = $request->get('query');

if (is_null($query))
    return redirect()->route($this->showRoute, [$date]);

$log     = $this->getLogOrFail($date);
$levels  = $this->logViewer->levelsNames();
$needles = explode(' ', $query);
$entries = $log->entries($level)->filter(function (LogEntry $value) use ($needles) {
    return Str::contains($value->header, $needles);
})->paginate($this->perPage);

return $this->view('show', compact('level', 'log', 'query', 'levels', 'entries'));
```

**Be careful with the spacing**

Reply:
Cool I didn't realize Str::containsAll() existed, good catch.